bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BRuntime.java

```diff
@@ -110,6 +111,35 @@ public void notifyFailure(ErrorValue error) {
         }
     }
 
+    public Object getSyncMethodInvokeResult(ObjectValue object, String methodName, Object... args) {
+        Function<?, ?> func = o -> object.call((Strand) (((Object[]) o)[0]), methodName, args);
+        Semaphore semaphore = new Semaphore(0);
+        final ErrorValue[] errorValue = new ErrorValue[1];
+        // Add 1 more element to keep null for add the strand later.
+        Object[] params = new Object[]{null, args};
+        FutureValue futureValue = scheduler.schedule(params, func, null, new CallableUnitCallback() {
+            @Override
+            public void notifySuccess() {
+                semaphore.release();
+            }
+
+            @Override
+            public void notifyFailure(ErrorValue error) {
+                errorValue[0] = error;
+                semaphore.release();
+            }
+        });
+        try {
+            semaphore.acquire();
```

Comment:
Shall we acquire the semaphore with the finite timeout? 
If the subsequent process failed or crashed, the ballerina thread blocks infinitely. 
Use `tryAcquire(xx, TimeUnit.SECONDS)` with a sufficient timeout period.